eth/handler.go

```diff
@@ -177,12 +188,13 @@ func NewProtocolManager(config *core.ChainConfig, fastSync bool, networkId int,
 	return manager, nil
 }
 
-func (pm *ProtocolManager) insertChain(blocks types.Blocks) (i int, err error) {
-	i, err = pm.blockchain.InsertChain(blocks)
+func (pm *ProtocolManager) insertChain(blocks types.Blocks) *core.ChainInsertResult {
+	res := pm.blockchain.InsertChain(blocks)
+	err := res.Error
 	if err != nil && pm.badBlockReportingEnabled && core.IsValidateError(err) {
-		go sendBadBlockReport(blocks[i], err)
+		go sendBadBlockReport(blocks[res.Index], err)
```

Comment:
Open idea on this. Do we actually ever use or need this? If not, we can get rid of this whole function.